src/index.js

```
@@ -79,170 +75,174 @@ const idyll = (inputPath, opts, cb) => {
-                  break;
-                case 'source':
-                  if (type === 'value') {
-                    source = value;
-                  }
-                  break;
-              };
-            })
-            if (source.endsWith('.csv')) {
-              parsed = Baby.parseFiles(path.join(DATA_FOLDER, source), { header: true });
-              data = parsed.data;
+
+          return acc;
+        },
+        {}
+      );
+    }
+
+    const getComponents = (ast) => {
+      const ignoreNames = ['var', 'data', 'meta', 'derived'];
+      const componentNodes = getNodesByName(s => !ignoreNames.includes(s), ast);
+
+      return componentNodes.reduce(
+        (acc, node) => {
+          const name = changeCase.paramCase(node[0]);
+
+          if (!acc[name]) {
+            if (inputConfig.components[name]) {
+              const compPath = path.parse(path.join(inputDirectory, inputConfig.components[name]));
+              acc[name] = path.join(path.relative(TMP_PATH, compPath.dir), compPath.base).replace(/\\/g, '/');
+            } else if (customComponents.indexOf(name + '.js') > -1) {
+              acc[name] = path.relative(TMP_PATH, path.join(CUSTOM_COMPONENTS_FOLDER, name)).replace(/\\/g, '/');
+            } else if (components.indexOf(name + '.js') > -1) {
+              acc[name] = path.relative(TMP_PATH, path.join(DEFAULT_COMPONENTS_FOLDER, name)).replace(/\\/g, '/');
             } else {
-              data = require(path.join(DATA_FOLDER, source));
+              try {
+                // npm modules are required via relative paths to support working with a locally linked idyll
+                const compPath = path.parse(resolve.sync(name, {basedir: inputDirectory}));
+                acc[name] = path.join(path.relative(TMP_PATH, compPath.dir), compPath.base).replace(/\\/g, '/');
+              } catch (err) {
+                if (node[0].toLowerCase() !== node[0]) throw new Error(`Component named ${node[0]} could not be found.`)
```

Won't that mean components that can't be found will fail silently?